### PR TITLE
Task slow mode swerve foc

### PR DESCRIPTION
**Pull request by Smogless**
added teleopswervelimited and enable FOC
AKit issues -> simulateJava doesnt work

### CODE REVIEW

**Comment by ChaoticChaosTheGreat** [comment]
Use last year code for fix : https://github.com/Team3256/FRC_Programming_2023/blob/main/src/main/java/frc/robot/Robot.java

**Comment by ChaoticChaosTheGreat** [comment]
someone merge this already

**Comment by 3LucasZ** [comment]
@ChaoticChaosTheGreat, please give us time to review your PR. In the meantime, please post a design document and a video of this working in simulation AND in real life. We can not merge this PR until that is done. Thank you for understanding.

File: src/main/deploy/pathplanner/paths/Red2-1Speaker-Mobility-L.path [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      },
+      "isLocked": false,
+      "linkedName": null
+    },
+    {
+      "anchor": {
+        "x": 11.221387350050021,
+        "y": 0.9017344140251753
+      },
+      "prevControl": {
```
Comment (3LucasZ) [inline src/main/deploy/pathplanner/paths/Red2-1Speaker-Mobility-L.path:22]: Why is a path included here? Please remove, this is unrelated to the issue being solved.

File: src/main/deploy/pathplanner/navgrid.json [hunk 2]
```diff
@@ -0,0 +1 @@
...se,true,true,true],[true,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,true,true],[true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true],[true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true],[true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true]]}
```
Comment (3LucasZ) [inline src/main/deploy/pathplanner/navgrid.json:1]: Unrelated to the issue

File: src/main/java/frc/robot/Constants.java [hunk 3]
```diff
@@ -76,6 +76,10 @@ public static final class FeatureFlags {
       public static final boolean kUsePrefs = true;
       public static final boolean kPitRoutineEnabled = false;
       public static final boolean kCanTestEnabled = false;
+      public static final double kSlowModeRatio = 0.4;
```
Comment (3LucasZ) [inline src/main/java/frc/robot/Constants.java:79]: make this kSlowModeSpeed instead of a ratio.

Comment (3LucasZ) [reply src/main/java/frc/robot/Constants.java:79]: I mean, kMaxSlowVelocity

File: src/main/java/frc/robot/Constants.java [hunk 4]
```diff
@@ -76,6 +76,10 @@ public static final class FeatureFlags {
...
+      public static final double kMaxAngularVelocity = 5;
```
Comment (3LucasZ) [inline src/main/java/frc/robot/Constants.java:80]: make a kMaxAngularSlowVelocity

Comment (3LucasZ) [reply src/main/java/frc/robot/Constants.java:80]: in addition to what is above

File: src/main/java/frc/robot/commands/TeleopSwerveLimited.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  }
+
+  @Override
+  public void execute() {
+    double yAxis = translationSup.getAsDouble() * kSlowModeRatio;
+    double xAxis = strafeSup.getAsDouble() * kSlowModeRatio;
+    double rAxis = rotationSup.getAsDouble() * kSlowModeRatio * kMaxAngularVelocity;
+
+    Translation2d translation = new Translation2d(yAxis, xAxis).times(kMaxVelocity);
+    swerveDrive.drive(translation, rAxis, robotCentricSup.getAsBoolean(), kOpenLoop);
```
Comment (3LucasZ) [inline src/main/java/frc/robot/commands/TeleopSwerveLimited.java:47]: won't review this section until it works IRL and in simulation, and there is a design doc linked to this PR (written in Google Docs or Notion)

**Review by 3LucasZ — CHANGES_REQUESTED** [review]

**Comment by ChaoticChaosTheGreat** [comment]
> @ChaoticChaosTheGreat, please give us time to review your PR. In the meantime, please post a design document and a video of this working in simulation AND in real life. We can not merge this PR until that is done. Thank you for understanding.

not my pr, this has just been open for too long

**Comment by ChaoticChaosTheGreat** [comment]
Also simulation of swerve does not work as of right now, will test on Monday if we have time. @Smogless

**Review by 3LucasZ — APPROVED** [review]
Looks good, works.

**Pull request merged**